packages/fluentui/react-northstar/src/components/Tree/hooks/useTree.ts

```diff
@@ -0,0 +1,174 @@
+  ]);
+
+  const getItemById = useGetItemById(flatTree);
+
+  const stableProps = useStableProps(options);
+
+  const toggleItemActive = React.useCallback(
+    (e: React.SyntheticEvent, idToToggle: string) => {
+      const item = getItemById(idToToggle);
+      if (!item || !item.hasSubtree) {
+        // leaf node does not have the concept of active/inactive
+        return;
+      }
+
+      setActiveItemIdsState(activeItemIds => {
+        let nextActiveItemIds: string[];
+        const isActiveId = activeItemIds.indexOf(idToToggle) !== -1;
+
+        if (isActiveId) {
+          nextActiveItemIds = _.without(activeItemIds, idToToggle);
+        } else {
+          nextActiveItemIds = [...activeItemIds, idToToggle];
+
+          if (options.exclusive) {
+            // remove active siblings, if any, from activeItemIds
+            const parent = getItemById(idToToggle)?.parent;
+            const activeSibling = getItemById(parent)?.childrenIds?.find(
+              id => id !== idToToggle && nextActiveItemIds.indexOf(id) >= 0,
+            );
+            if (activeSibling != null) {
+              nextActiveItemIds = _.without(nextActiveItemIds, activeSibling);
+            }
+          }
+        }
+
+        _.invoke(stableProps.current, 'onActiveItemIdsChange', e, {
+          ...stableProps.current,
+          activeItemIds: nextActiveItemIds,
+        });
+
+        return nextActiveItemIds;
+      });
+    },
+    [getItemById, options.exclusive, setActiveItemIdsState, stableProps],
+  );
+
+  const expandSiblings = React.useCallback(
+    (e: React.KeyboardEvent, focusedItemId: string) => {
+      if (options.exclusive) {
+        return;
+      }
+
+      const focusedItem = getItemById(focusedItemId);
+      if (!focusedItem) {
+        return;
+      }
+
+      const parentItem = getItemById(focusedItem?.parent);
+      const siblingsIds = parentItem?.childrenIds;
+
+      if (!siblingsIds) {
+        return;
+      }
+
+      setActiveItemIdsState(activeItemIds => {
+        const nextActiveItemIds = _.uniq(activeItemIds.concat(siblingsIds));
+        _.invoke(stableProps.current, 'onActiveItemIdsChange', e, {
+          ...stableProps.current,
+          activeItemIds: nextActiveItemIds,
+        });
+        return nextActiveItemIds;
+      });
+    },
+    [getItemById, options.exclusive, setActiveItemIdsState, stableProps],
+  );
+
+  // Maintains stable collection of refs to avoid unnecessary React context updates
+  const nodes = React.useRef<Record<string, HTMLElement>>({});
+  const registerItemRef = React.useCallback((id: string, node: HTMLElement) => {
+    nodes.current[id] = node;
+  }, []);
+  const getItemRef = React.useCallback((id): HTMLElement => nodes.current[id], []);
+
+  // can be used for keyboard navigation ===
+  const focusItemById = React.useCallback(
+    (id: string) => {
+      if (id == null) {
+        return;
+      }
+      const itemRef = getItemRef(id);
+      if (itemRef == null) {
```

Comment:
There should an ESLint rule that prevents non-strict checks, I  would rather go with as it's safer:

```jsx
      if (itemRef instanceof HTMLElement) {
        if (getItemById(id)?.hasSubtree) {
          itemRef.focus();
        } else {
          // when node is leaf, need to focus on the inner treeTitle
          (itemRef.firstElementChild as HTMLElement)?.focus();
        }
      }
```